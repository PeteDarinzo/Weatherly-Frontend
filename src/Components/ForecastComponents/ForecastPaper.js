import React from "react";
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import Box from '@mui/material/Box';

const styles = {
  media: {
    height: 100,
    width: 100
  }
}

const ForecastPaper = ({ day, date, description, min, max, feelsLike, icon, units, compatibility }) => {
  return (
    <Paper
      elevation={12}
      style={{
        padding: 8,
        border: "1px solid black",
        // justifyContent: "center",
        // alignItems: "center",
        textAlign: "center",
      }}
    >
      <Typography>{day}</Typography>
      <Typography>{date}</Typography>
      <Typography>
        {description}
      </Typography>
      <Box
        component="img"
        sx={{
          height: 75,
          width: 75,
          maxHeight: { xs: 233, md: 167 },
          maxWidth: { xs: 350, md: 250 },
        }}
        alt={description}
        src={`http://openweathermap.org/img/wn/${icon}@2x.png`}
        style={{
          backgroundColor: "lightskyblue"
        }}
      />
      <Typography>
        Min: {min} {units}
      </Typography>
      <Typography>
        Max: {max} {units}
      </Typography>
      <Typography>
        Feels Like: {feelsLike} {units}
      </Typography>
      <Typography>
        Good day to watch? {compatibility}
      </Typography>
    </Paper >
  );
}

export default ForecastPaper;