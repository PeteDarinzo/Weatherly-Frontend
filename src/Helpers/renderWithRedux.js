import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}