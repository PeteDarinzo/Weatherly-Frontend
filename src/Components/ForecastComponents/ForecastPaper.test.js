import ForecastPaper from "./ForecastPaper";
import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";

it("renders forecast paper without crashing", () => {
  renderWithRedux(
    <MemoryRouter>
      <ForecastPaper day="mon" date="1/1" description="cloudy" feelsLike="55" icon="" units="F" compatibility={1} />
    </MemoryRouter>
  );
});


it("matches the forecast paper snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <ForecastPaper day="mon" date="1/1" description="cloudy" feelsLike="55" icon="" units="F" compatibility={1} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
