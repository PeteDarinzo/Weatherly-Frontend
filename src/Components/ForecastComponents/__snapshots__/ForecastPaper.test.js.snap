// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the forecast paper snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation12 css-15s4tt4-MuiPaper-root"
    style="padding: 8px; border: 1px solid black; text-align: center;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1f7e215-MuiTypography-root"
        >
          mon
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1v1tjum-MuiTypography-root"
        >
          1/1
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="makeStyles-box-2 MuiBox-root css-165t1cv"
          />
          <div
            class="makeStyles-box-2 MuiBox-root css-165t1cv"
          />
          <div
            class="makeStyles-box-2 MuiBox-root css-1u9s4zi"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
      >
        <img
          alt="cloudy"
          class="MuiBox-root css-46jsvw"
          src="http://openweathermap.org/img/wn/@2x.png"
          style="background-color: lightskyblue;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-qjpmex-MuiTypography-root"
        >
          55 Â°F
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ee8j7p-MuiTypography-root"
        >
          cloudy
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
